src/main/java/com/powsybl/openloadflow/OpenSecurityAnalysis.java

```diff
@@ -0,0 +1,474 @@
+        AcloadFlowEngine engine = new AcloadFlowEngine(network, acParameters);
+        AcLoadFlowResult preContingencyLoadFlowResult = engine.run();
+        boolean preContingencyComputationOk = preContingencyLoadFlowResult.getNewtonRaphsonStatus() == NewtonRaphsonStatus.CONVERGED;
+        List<LimitViolation> preContingencyLimitViolations = new ArrayList<>();
+        LimitViolationsResult preContingencyResult = new LimitViolationsResult(preContingencyComputationOk, preContingencyLimitViolations);
+
+        // only run post-contingency simulations if pre-contingency simulation is ok
+        List<PostContingencyResult> postContingencyResults = new ArrayList<>();
+        if (preContingencyComputationOk) {
+            detectViolations(network.getBranches().stream(), network.getBuses().stream(), preContingencyLimitViolations);
+
+            LOGGER.info("Save pre-contingency state");
+
+            // save base state for later restoration after each contingency
+            Map<LfBus, BusState> busStates = getBusStates(network.getBuses(), engine);
+
+            // start a simulation for each of the contingency
+            Iterator<LfContingency> contingencyIt = contingencies.iterator();
+            while (contingencyIt.hasNext()) {
+                LfContingency lfContingency = contingencyIt.next();
+
+                PostContingencyResult postContingencyResult = runPostContingencySimulation(network, engine, lfContingency);
+                postContingencyResults.add(postContingencyResult);
+
+                if (contingencyIt.hasNext()) {
+                    LOGGER.info("Restore pre-contingency state");
+
+                    double mismatch = 0;
+                    for (LfBus bus : lfContingency.getBuses()) {
+                        mismatch += bus.getGenerationTargetP() - bus.getLoadTargetP();
+                    } //TODO: if mismatch different than zero, start with a slack distribution.
+
+                    // restore base state
+                    restoreBusStates(busStates, engine);
+                }
+            }
+        }
+
+        return new SecurityAnalysisResult(preContingencyResult, postContingencyResults);
+    }
+
+    /**
+     * Get the map of the states of given buses, indexed by the bus itself
+     * @param buses the bus for which the state is returned
+     * @param engine
+     * @return the map of the states of given buses, indexed by the bus itself
+     */
+    private Map<LfBus, BusState> getBusStates(List<LfBus> buses, AcloadFlowEngine engine) {
+        Map<LfBus, BusState> result = new HashMap<>(buses.size());
+        buses.forEach(b -> {
+            Map<String, Double> generatorsTargetP = b.getGenerators().stream().collect(
+                Collectors.toMap(LfGenerator::getId, LfGenerator::getTargetP));
+            BusState busState = new BusState(b.getV(), b.getAngle(), b.getLoadTargetP(), generatorsTargetP);
+            result.put(b, busState);
+            switchPqPv(b, engine);
```

Comment:
I think that I made a mistake here: we don't have to call this function ````switchPqPv```` at this place. We have to save the information of which buses are PQ now and where PV at the beginning of the LF and the associated targetQ.